packages/react-reconciler/src/ReactProfilerTimer.js

```diff
@@ -77,9 +80,78 @@ function stopProfilerTimerIfRunningAndRecordDelta(
   }
 }
 
+function recordLayoutEffectDuration(fiber: Fiber): void {
+  if (!enableProfilerTimer) {
+    return;
+  }
+
+  if (layoutEffectStartTime >= 0) {
+    const elapsedTime = now() - layoutEffectStartTime;
+
+    layoutEffectStartTime = -1;
+
+    // Store duration on the next nearest Profiler ancestor.
+    let parentFiber = fiber.return;
+    while (parentFiber !== null) {
+      if (parentFiber.tag === Profiler) {
+        const parentStateNode = parentFiber.stateNode;
+        parentStateNode.effectDuration += elapsedTime;
+        break;
+      }
+      parentFiber = parentFiber.return;
+    }
+  }
+}
+
+function recordPassiveEffectDuration(fiber: Fiber): void {
+  if (!enableProfilerTimer) {
+    return;
+  }
+
+  if (passiveEffectStartTime >= 0) {
+    const elapsedTime = now() - passiveEffectStartTime;
+
+    passiveEffectStartTime = -1;
+
+    // Store duration on the next nearest Profiler ancestor.
+    let parentFiber = fiber.return;
+    while (parentFiber !== null) {
+      if (parentFiber.tag === Profiler) {
+        const parentStateNode = parentFiber.stateNode;
+        if (parentStateNode !== null) {
```

Comment:
This check is something that we wouldn't need to do if we eagerly flushed passive destroy functions during unmount.